From dc3ab3a34374f19d3d6a833aaad93f90d2471a5f Mon Sep 17 00:00:00 2001
From: Erik Faye-Lund <erik.faye-lund@collabora.com>
Date: Fri, 30 Jun 2023 15:00:42 +0000
Subject: [PATCH] HACK: panvk: report support for vk 1.1

A lot of applications require VK 1.1... so let's try to report that.

VK 1.1 requires multiview, so let's pretend that we support that as
well...
---
 src/panfrost/vulkan/panvk_device.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/panfrost/vulkan/panvk_device.c b/src/panfrost/vulkan/panvk_device.c
index a823c5b617d..f9ac4e97c27 100644
--- a/src/panfrost/vulkan/panvk_device.c
+++ b/src/panfrost/vulkan/panvk_device.c
@@ -121,7 +121,7 @@ static const struct debug_control panvk_debug_options[] = {
 #define PANVK_USE_WSI_PLATFORM
 #endif
 
-#define PANVK_API_VERSION VK_MAKE_VERSION(1, 0, VK_HEADER_VERSION)
+#define PANVK_API_VERSION VK_MAKE_API_VERSION(0, 1, 1, VK_HEADER_VERSION)
 
 VkResult
 panvk_EnumerateInstanceVersion(uint32_t *pApiVersion)
@@ -433,7 +433,7 @@ panvk_GetPhysicalDeviceFeatures2(VkPhysicalDevice physicalDevice,
       .uniformAndStorageBuffer16BitAccess = false,
       .storagePushConstant16 = false,
       .storageInputOutput16 = false,
-      .multiview = false,
+      .multiview = true,
       .multiviewGeometryShader = false,
       .multiviewTessellationShader = false,
       .variablePointersStorageBuffer = true,
-- 
GitLab

